### ISSUE TITLE
cargo --version is not same as rustc --version for rust-1.56.1 

### ISSUE THREAD

**Issue by sibskull** [rust-lang/cargo#10122]
### Problem

```# rustc --version
rustc 1.56.1
# cargo --version
cargo 1.56.0```

I guess this is wrong because there is note at https://github.com/rust-lang/cargo/blob/7f08ace4f1305de7f3b1b0e2f765911957226bd4/src/cargo/lib.rs#L160

Rust 1.56.1 is built from source tarball.

### Steps

_No response_

### Possible Solution(s)

_No response_

### Notes

_No response_

### Version

```text
cargo 1.56.0
release: 1.56.0
```

**Comment by Eh2406** [comment]
I think that the Cargo version changes when there are changes to Cargo. If the patch release of Rust did not need a change to Cargo then Cargos version would not change.

**Comment by ehuss** [comment]
I have posted a fix for this at #10178.
